rook/scripts/funcs.sh

```diff
@@ -0,0 +1,400 @@
+      logJob "${1}" "${2}" "${4}"
+    fi
+}
+
+# Args:
+#   1. namespace
+#   2. name
+#   3. cluster
+function logJob {
+    writeLog "${1}" "${2}" "Job" "${3}"
+    writeEvent "${1}" "${2}" "Job" "${3}"
+    writeEvent "${1}" "${2}" "Pod" "${3}"
+}
+
+LOGSFOLDER="logs"
+EVENTSFOLDER="events"
+
+# This function writes logs to file for specified <kind>
+# Args:
+#   1. namespace
+#   2. name
+#   3. kind
+#   4. cluster
+function writeLog {
+    if [[ -z "$PIPELINE" ]]; then
+        return
+    fi
+
+    NAMESPACE=$1
+    NAME=$2
+    KIND=$3
+    CLUSTER=$4
+    NAMES=$(kubectl get "$KIND" -n "$NAMESPACE" -o custom-columns=NAME:.metadata.name | grep "$NAME" | tail -n +1)
+    mapfile -t NAMESLIST <<< "$NAMES"
+
+    mkdir -p "./$LOGSFOLDER/$CLUSTER/$KIND/$NAMESPACE"
+    for NAME in "${NAMESLIST[@]}"
+    do
+        FILE="./$LOGSFOLDER/$CLUSTER/$KIND/$NAMESPACE/$NAME.log"
+        if [[ ! -f "$FILE" ]]; then
+            touch "$FILE"
+            kubectl -n "$NAMESPACE" logs "$KIND"/"$NAME" --all-containers=true > "$FILE" 2>&1
+        fi
+    done
+}
+
+# This function writes events to file for specified <kind>
+# Args:
+#   1. namespace
+#   2. name
+#   3. kind
+#   4. cluster
+function writeEvent {
+    if [[ -z "$PIPELINE" ]]; then
+        return
+    fi
+
+    NAMESPACE=$1
+    NAME=$2
+    KIND=$3
+    CLUSTER=$4
+    NAMES=$(kubectl get "$KIND" -n "$NAMESPACE" -o custom-columns=NAME:.metadata.name | grep "$NAME" | tail -n +1)
+    mapfile -t NAMESLIST <<< "$NAMES"
+
+    mkdir -p "./$EVENTSFOLDER/$CLUSTER/$KIND/$NAMESPACE"
+    for NAME in "${NAMESLIST[@]}"
+    do
+        FILE="./$EVENTSFOLDER/$CLUSTER/$KIND/$NAMESPACE/$NAME.event"
+        if [[ ! -f "$FILE" ]]; then
+            touch "$FILE"
+            DATA=$(kubectl get event -n "${NAMESPACE}" --field-selector involvedObject.kind="${KIND}",involvedObject.name="${NAME}" -o json)
+            MESSAGES=$(echo "${DATA}" | jq -r '.items | map(.message) | .[]')
+            echo "$MESSAGES" > "$FILE"
+        fi
+    done
+}
+
+# Args:
+#   1. Name of cephcluster
+function testCephCluster {
+  jsonData=$(kubectl -n rook-ceph get cephclusters.ceph.rook.io "${1}" -ojson)
+  health=$(echo "${jsonData}" | jq -r '.status.ceph.health')
+  phase=$(echo "${jsonData}" | jq -r '.status.phase')
+  if [[ ! $health = "HEALTH_OK" ]]; then
+    echo "Cluster health is not ok: $health ❌"; FAILURES=$((FAILURES+1))
+    DEBUG_OUTPUT+=$(echo "${jsonData}" | jq '.status.ceph.details')
+  else
+    echo "Cluster health is ok ✔"
+  fi
+  if [[ ! $phase = "Ready" ]]; then
+    echo "Cluster phase is ready: $phase ❌"; FAILURES=$((FAILURES+1))
+  else
+    echo "Cluster phase is ready ✔"; SUCCESSES=$((SUCCESSES+1))
+  fi
+}
+
+function testNumberOfOSDs {
+  numberOfOSDs=$(kubectl -n rook-ceph get deployments.apps -o=jsonpath='{.items[*].metadata.name}' |  xargs -n 1 | grep -c rook-ceph-osd)
+  numberOfWorkerNodes=$(kubectl get nodes -o=jsonpath='{.items[*].metadata.name}' |  xargs -n 1 | grep -c worker)
```

Comment:
This shouldn't be equal to the number of worker nodes, but rather equal to the number of storage nodes

```suggestion
  numberOfWorkerNodes=$(kubectl get nodes -o=jsonpath='{.items[*].metadata.name}' |  xargs -n 1 | grep -cP "storage-\d+$")
```

Reply:
Would it be fine to have a check if there are 0 storage nodes, then it uses the number of workers? I'm thinking for my devcluster personally and the pipeline which don't have dedicated storage nodes. Or would we prefer some other solution? :thinking:

Reply:
Right, yea I guess that would work :point_up: